tool/src/org/antlr/v4/Tool.java

```diff
@@ -397,7 +397,7 @@ public void processNonCombinedGrammar(Grammar g, boolean gencode) {
 
 		if ( generate_ATN_dot ) generateATNs(g);
 
-		if ( g.tool.getNumErrors()==0 ) generateInterpreterData(g);
+		if (gencode && g.tool.getNumErrors()==0 ) generateInterpreterData(g);
```

Comment:
ok, looking good but do we need this tweak for gencode? I'm worried someone might rely on it.

Reply:
This seems like a safe change to me– code is still generated when executed form the cli, but won't generate code when executed programmatically. I think there was an issue with it writing files in a readonly environment (e.g., deployed as a Lambda).

Reply:
Thanks for looking at it!